TLM/TLM/Manager/Impl/LaneConnectionManagerData/ConnectionDataBase.cs

```diff
@@ -0,0 +1,186 @@
+namespace TrafficManager.Manager.Impl.LaneConnectionManagerData {
+    using CSUtil.Commons;
+    using System.Collections.Generic;
+    using System.Diagnostics;
+    using TrafficManager.Util;
+    using TrafficManager.Util.Extensions;
+
+    internal class ConnectionDataBase : Dictionary<LaneEnd, LaneConnectionData[]> {
+        internal bool IsConnectedTo(uint sourceLaneId, uint targetLaneId, ushort nodeId) =>
+            IsConnectedTo(sourceLaneId, targetLaneId, sourceLaneId.ToLane().IsStartNode(nodeId));
+
+        /// <param name="sourceStartNode">start node for the segment of the source lane</param>
+        internal bool IsConnectedTo(uint sourceLaneId, uint targetLaneId, bool sourceStartNode) {
+            var key = new LaneEnd(sourceLaneId, sourceStartNode);
+            if (this.TryGetValue(key, out var targets)) {
+                int n = targets.Length;
+                for (int i = 0; i < n; ++i) {
+                    if (targets[i].Enabled && targets[i].LaneId == targetLaneId) {
+                        return true;
+                    }
+                }
+            }
+            return false;
+        }
+
+        /// <summary>
+        /// Adds a connection from source to target lane at the give node
+        /// </summary>
+        internal void ConnectTo(uint sourceLaneId, uint targetLaneId, ushort nodeId) {
+            AddConnection(sourceLaneId, targetLaneId, nodeId, true); //forward
+            AddConnection(targetLaneId, sourceLaneId, nodeId, false); //backward
+        }
+
+        /// <summary>
+        /// makes a uni-directional connection
+        /// </summary>
+        /// <param name="enable"><c>true</c> for forward/bi-directional connection,
+        /// <c>false</c> for backward connection</param>
+        private void AddConnection(uint sourceLaneId, uint targetLaneId, ushort nodeId, bool enable) {
```

Comment:
> makes a uni-directional connection

_... or makes it bi-directional if exist_? Am I right?

Reply:
2 uni-directional connections means bidirectional.

there is no such thing as a bi-directional lane connection. its stored as two uni-directional lane connections.
Enabled = real connection
disabled = not real lane connection. its just there as a hint to help finding the Enabled one.